src/main.rs

```diff
@@ -0,0 +1,295 @@
+use clap::{App, Arg, SubCommand};
+use serde::{Deserialize, Serialize};
+use std::error::Error;
+
+static PARENT_FOLDER: &str = ".graft.lock";
+static TMP_FOLDER: &str = "tmp";
+static CONFIG_FILE: &str = "graft.txt";
+static LOCK_FILE: &str = "conflict.lock";
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Repo {
+    upstream_path: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Config {
+    repos_to_fetch: Vec<Repo>,
+}
+
+fn read_config<P: AsRef<std::path::Path>>(path: P) -> Result<Config, Box<dyn Error>> {
+    let file = std::fs::File::open(path)?;
+    let reader = std::io::BufReader::new(file);
+    let config = serde_json::from_reader(reader)?;
+    Ok(config)
+}
+
+fn write_file_content(content: String, file_path: &str) -> Result<(), Box<dyn Error>> {
+    match std::path::Path::new(file_path).parent() {
+        Some(p) => {
+            std::fs::create_dir_all(p);
+        }
+        None => {}
+    }
+    let mut out = std::fs::File::create(file_path)
+        .expect(format!("ERROR: failed to create file: {:?}", file_path).as_str());
```

Comment:
In a Result function there is no need for `.expect()`, which crashes the application.